controllers/workspace/status.go

```diff
@@ -62,6 +63,7 @@ var healthHttpClient = &http.Client{
 // Parameters for result and error are returned unmodified, unless error is nil and another error is encountered while
 // updating the status.
 func (r *DevWorkspaceReconciler) updateWorkspaceStatus(workspace *dw.DevWorkspace, logger logr.Logger, status *currentStatus, reconcileResult reconcile.Result, reconcileError error) (reconcile.Result, error) {
+	updateMetricsForPhase(workspace, status.phase, logger)
```

Comment:
I assume metrics will be more precise if we move after check that status update error is nil

Reply:
Move to only update metrics if the corresponding status update succeeds.